Update session handler and load current user

marcusesa opened the pull request:
## Description

Update session handler from **native handler** to **redis handler** and allow the loading of current user
based on _UsernamePasswordToken_.

## Motivation and context

In order to know who is the user logged in qedu-hub, we added some new configurations to load the current user using session content storage in Redis.

## Main Changes

- Install and register SncRedisBundle and Predis.
- Update session handler from native to redis.
- Configure user encoder.
- Configure database user loader.
- Add user entity.
- Add user repository.

## Reference

- [Symfony documentation](https://symfony.com/doc/current/security/entity_provider.html#security-config-entity-provider).
- [SncRedisBundle documentation.](https://github.com/snc/SncRedisBundle)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Screenshots 

<img width="1280" alt="screen shot 2017-08-30 at 8 07 35 pm" src="https://user-images.githubusercontent.com/1117674/29898966-08722ac0-8dbf-11e7-9095-e8f465f697d3.png">
<img width="1280" alt="screen shot 2017-08-30 at 8 07 59 pm" src="https://user-images.githubusercontent.com/1117674/29898967-08728218-8dbf-11e7-9aa3-87b581884cb2.png">

marcusesa committed:
Update session handler and load current user

Update session handler from native to redis, and load the current user
based on UsernamePasswordToken.

Main Changes:

- Install and register SncRedisBundle and Predis.
- Update session handler from native to redis.
- Configure user encoder.
- Configure database user loader.
- Add user entity.
- Add user repository.

marcusesa committed:
Fix Code Style in entity user

marcusesa committed:
Add redis_host key in parameters.yml.dist file

marcusesa committed:
Add getters methods to name and surname property in Entity/User

gilsonqedu reviewed (commented):
That's ok this changes.

marcusesa committed:
Add user dropdown menu or signup/sign links based on user status

If there is an user logged we show user dropdown menu otherwise
we show signup/sign links in navbar.

marcusesa committed:
Add UserTest

Add UserTest, only to check if User implements UserInterface and
Serializable.

marcusesa committed:
Add config/parameters.yml to linked files in Capistrano script

marcusesa committed:
Update asset url

gilsonqedu commented on app/config/config.yml:
```diff
@@ -67,3 +66,13 @@ swiftmailer:
+        session:
+            type: predis
+            alias: session
+            dsn: 'redis://%redis_host%'
```
How get the value of redis_host? Variable environment?

marcusesa replied:
Actually is from the file [parameters.yml](https://github.com/QEdu/qedu-hub/blob/ba62ebae3f4e91a24d77c338c959fe082b1f036a/app/config/parameters.yml.dist#L21), but in the future would be nice have it in environment variables.

marcusesa committed:
Add default ttl in redis sessions
